src/main/java/edu/cornell/kfs/fp/service/impl/RecurringDisbursementVoucherDocumentServiceImpl.java

```diff
@@ -311,7 +339,72 @@ private RecurringDisbursementVoucherPDPStatus buildRecurringDisbursementVoucherP
+            Note paymentGroupNote = buildNoteBase();
+            paymentGroupNote.setNoteText(noteText + StringUtils.join(setOfStrings, ", "));;
+            recurringDV.addNote(paymentGroupNote);
+            getBusinessObjectService().save(recurringDV);
+        }
+    }
+    
+    private Note buildNoteBase() {
+        Note noteBase = new Note();
+        noteBase.setAuthorUniversalIdentifier(GlobalVariables.getUserSession().getPerson().getPrincipalId());
+        noteBase.setNotePostedTimestampToCurrent();
+        return noteBase;
+    }
+    
+    @Override
+    public boolean isPaymentCancelable(Person user, String paymentDetailStatus) {
+        return paymentDetailStatus.equalsIgnoreCase(PdpConstants.PaymentStatusCodes.OPEN) 
+                && getRecurringDisbursementVoucherPaymentMaintenanceService().hasCancelPermission(user);
+    }
+    
+    @Override
+    public Set<String> cancelDisbursementVouchers(RecurringDisbursementVoucherDocument recurringDisbursementVoucherDocument, final String cancelMessage) {
+        Set<String> canceledDVs = new HashSet<String>();
+        for (RecurringDisbursementVoucherDetail detail : recurringDisbursementVoucherDocument.getRecurringDisbursementVoucherDetails()) {
+            CuDisbursementVoucherDocument dv = retrieveCUDisbursementVoucherDocument(detail);
+            if (isDVCancelable(dv)) {
+                cancelDVAsSystemUser(cancelMessage, dv);
+                canceledDVs.add(dv.getDocumentNumber());
+            }
+        }
+        noteChangeOnRecurringDV(recurringDisbursementVoucherDocument, "The following Disbursement Vouchers were canceled: ", canceledDVs);
+        return canceledDVs;
+    }
+
+    private void cancelDVAsSystemUser(final String cancelMessage, CuDisbursementVoucherDocument dv) {
+        final String docuemntNumber = dv.getDocumentNumber();
+        try {
+            GlobalVariables.doInNewGlobalVariables(new UserSession(KFSConstants.SYSTEM_USER), new Callable<Object>() {
+                @Override
+                public Object call() throws WorkflowException {
+                    CuDisbursementVoucherDocument disbursementVoucher = (CuDisbursementVoucherDocument) getDocumentService().getByDocumentHeaderId(docuemntNumber);
+                    getDocumentService().cancelDocument(disbursementVoucher, cancelMessage);
+                    return null;
+                }
+            });
+        } catch (Exception e) {
+            throw new RuntimeException("cancelDVAsSystemUser() Unable to cancel DV: " + dv.getDocumentNumber(), e);
+        }
+    }
+    
+    private boolean isDVCancelable(CuDisbursementVoucherDocument dv) {
+        return getRecurringDisbursementVoucherPaymentMaintenanceService().hasCancelPermission(GlobalVariables.getUserSession().getPerson()) && 
+                StringUtils.equalsIgnoreCase(dv.getDocumentHeader().getWorkflowDocument().getStatus().getLabel(), DocumentStatus.SAVED.getLabel());
```

Comment:
Another spot where the "&&" could be moved to the second line.
Also, if you're comparing KEW document statuses and not application document statuses, then I would suggest comparing the status codes instead of the status labels.